include/seqan3/io/sequence_file/format_genbank.hpp

```diff
@@ -0,0 +1,266 @@
+     * \{
+     */
+     //!\brief Default constructor is explicitly defaulted, you need to provide a stream or file name.
+    sequence_file_format_genbank() = default;
+    //!\brief Copy construction is explicitly deleted, because you can't have multiple access to the genbank file.
+    sequence_file_format_genbank(sequence_file_format_genbank const &) = delete;
+    //!\brief Copy assignment is explicitly deleted, because you can't have multiple access to the genbank file.
+    sequence_file_format_genbank & operator=(sequence_file_format_genbank const &) = delete;
+    //!\brief Move construction is defaulted.
+    sequence_file_format_genbank(sequence_file_format_genbank &&) = default;
+    //!\brief Move assignment is defaulted.
+    sequence_file_format_genbank & operator=(sequence_file_format_genbank &&) = default;
+    //!\}
+
+    //!\brief The valid file extensions for this format; note that you can modify this value.
+    static inline std::vector<std::string> file_extensions
+    {
+        { "genbank" },
+        { "gb" },
+        { "gbk" },
+    };
+
+    //!\copydoc SequenceFileInputFormat::read
+    template <typename stream_type,     // constraints checked by file
+              typename seq_legal_alph_type, bool seq_qual_combined,
+              typename seq_type,        // other constraints checked inside function
+              typename id_type,
+              typename qual_type>
+    void read(stream_type                                                            	& stream,
+              sequence_file_input_options<seq_legal_alph_type, seq_qual_combined> const & options,
+              seq_type                                                                  & sequence,
+              id_type                                                                   & id,
+              qual_type                                                                 & SEQAN3_DOXYGEN_ONLY(qualities))
+    {
+        auto stream_view = std::ranges::subrange<decltype(std::istreambuf_iterator<char>{stream}),
+                                                 decltype(std::istreambuf_iterator<char>{})>
+                           {std::istreambuf_iterator<char>{stream},
+                            std::istreambuf_iterator<char>{}};
+        auto stream_it = std::ranges::begin(stream_view);
+
+        std::string idbuffer;
+        std::ranges::copy(stream_view | view::take_until_or_throw(is_cntrl || is_blank),
+                          std::back_inserter(idbuffer));
+
+        if (idbuffer != "LOCUS")
+            throw parse_error{"An entry has to start with the code word LOCUS."};
+
+        //ID
+        if constexpr (!detail::decays_to_ignore_v<id_type>)
+        {
+            if (options.embl_genbank_complete_header)
+            {
+                while (idbuffer != "ORIGIN")
+                {
+                        std::ranges::copy(idbuffer  | view::char_to<value_type_t<id_type>>,
```

Comment:
hm.. you always copy twice for each line (once into buffer and once into id). 
I would propose the following: read directly into the id (no buffer needed at all), the while condition will then be 
`while (std::ranges::equal(subrange{id.end() - 7, id.end()}, std::string{ORIGIN\n})` (we would need to require CommonRange and RandomAccessRange but everybody should use std::string for ids anyway..)
and once ORIGIN is reached it can be deleted again (requires sequence_container_concept..).

@h-2  What do you think?
Normal `until char` stuff doesn't work nicely because we have key words instead of key chars.

Reply:
Why should I use the subrange, instead of the stream until a blank appears?